apps/mobile/src/features/send-refactor/forms/btc/use-btc-form.ts

```diff
@@ -0,0 +1,82 @@
+import { useForm } from 'react-hook-form';
+
+import { useToastContext } from '@/components/toast/toast-context';
+import {
+  type BtcFormSchema,
+  useBtcSendFormSchema,
+} from '@/features/send-refactor/forms/btc/btc-form-schema';
+import { useCalculateBtcMaxSpend } from '@/features/send-refactor/hooks/use-calculate-btc-max-spend';
+import { useSendMax } from '@/features/send-refactor/hooks/use-send-max';
+import { useSendNavigation } from '@/features/send-refactor/navigation';
+import { btcFormValuesToPsbtHex } from '@/features/send-refactor/utils';
+import { Account } from '@/store/accounts/accounts';
+import { useBitcoinAccounts } from '@/store/keychains/bitcoin/bitcoin-keychains.read';
+import { useSettings } from '@/store/settings/settings';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { t } from '@lingui/macro';
+
+import { AverageBitcoinFeeRates } from '@leather.io/models';
+import type { Utxo } from '@leather.io/query';
+
+interface UseBtcFormProps {
+  account: Account;
+  feeRates: AverageBitcoinFeeRates;
+  utxos: Utxo[];
+}
+
+export function useBtcForm({ account, feeRates, utxos }: UseBtcFormProps) {
+  const { accountIndex, fingerprint } = account;
+  const { navigate } = useSendNavigation();
+  const { displayToast } = useToastContext();
+  const { networkPreference } = useSettings();
+  const { nativeSegwit } = useBitcoinAccounts().accountIndexByPaymentType(
+    fingerprint,
+    accountIndex
+  );
+  const calculateMaxBtcSpend = useCalculateBtcMaxSpend(feeRates, utxos);
+  const schema = useBtcSendFormSchema({
+    networkMode: networkPreference.chain.bitcoin.mode,
+    calculateBtcMaxSpend: calculateMaxBtcSpend,
+  });
+  const form = useForm<BtcFormSchema>({
+    resolver: zodResolver(schema),
+    mode: 'onTouched',
+    defaultValues: {
+      amount: '0',
+      recipient: '',
+      isSendingMax: false,
+      feeRate: feeRates.halfHourFee.toNumber(),
+    },
+  });
+  const maxSpend = calculateMaxBtcSpend({
+    recipient: form.watch('recipient'),
+    feeRate: form.watch('feeRate'),
+  });
+  const { onSetMax } = useSendMax(maxSpend.spendableBitcoin, form);
+
+  const handleSubmit = form.handleSubmit(values =>
+    btcFormValuesToPsbtHex(
+      values,
+      nativeSegwit.derivePayer({ addressIndex: 0 }),
+      utxos,
+      networkPreference.chain.bitcoin.mode
+    )
+      .then((psbtHex: string) => navigate('approval', { hex: psbtHex }))
+      .catch(() =>
+        displayToast({
```

Comment:
Maybe we can throw the error here and it can be caught by `formatBitcoinError` / `formatStacksError` ?

Reply:
Generally not a good idea to throw errors in the render cycle. What would be the benefit of that?

Reply:
It's just to keep consistency with how we are handling all the other errors and to avoid repeating `displayToast`. 

Maybe we need to do more refactoring to deprecate `formatBitcoinError` instead?

Reply:
I'm not sure I'm following. There just two `displayToast` occurences in the entire flow, both are for very unlikely unhandled exceptions. Most of the validation and error display is done through `zod` + RHF, the form by design doesn't allow submission when errors are present. This is just for an unlikely case of final TX somehow being invalid. I've excluded analytics from this PR to avoid growing it even larger, but we'll be also tracking those events.

Reply:
The pattern we inherited was that - packages throw and error and it gets caught and displayed by `formatBitcoinError` in `use-send-form-btc`:
```
 catch (e) {
        if (e instanceof BitcoinError) {
          displayToast({ title: formatBitcoinError(e.message), type: 'error' });
        } else {
          displayToast({
            title: t({ id: 'something-went-wrong', message: 'Something went wrong' }),
            type: 'error',
          });
        }
```

I agree it's not good to throw errors in the render cycle but am suggesting that we keep consistent and do that until we can refactor and improve. 

It's kind of hard to tell the changes here as you aren't deleting the old code yet like - `use-send-form-btc`